assignments/assignment7/faulty-oops.md

```diff
@@ -0,0 +1,54 @@
+# Kernel Oops Analysis - Assignment 7
+
+## **0. Run the command**
+- **echo “hello_world” > /dev/faulty**
+
+## **1. Kernel Oops Details**
+- **Error Type:** NULL Pointer Dereference
+- **Faulting Address:** `0x0000000000000000` (NULL)
+- **Process Affected:** `sh` (Shell)
+- **Error Code (ESR):** `0x96000045`
+- **Function Causing the Fault:** `faulty_write+0x10/0x20 [faulty]`
+- **Call Trace:**
+  ```
+  faulty_write+0x10/0x20 [faulty]
+  ksys_write+0x74/0x110
+  __arm64_sys_write+0x1c/0x30
+  invoke_syscall+0x54/0x130
+  el0_svc_common.constprop.0+0x44/0xf0
+  do_el0_svc+0x2c/0xc0
+  el0_svc+0x2c/0x90
+  el0t_64_sync_handler+0xf4/0x120
+  el0t_64_sync+0x18c/0x190
+  ```
+
+## **2. Cause of the Oops**
+The kernel oops was triggered by a **NULL pointer dereference** in the `faulty_write` function of the **faulty** module.
+
+- The crash happened because `faulty_write` attempted to write to memory at address `0x0000000000000000`, which is an **invalid memory access**.
+- The function did not check whether a pointer was `NULL` before dereferencing it.
```

Comment:
objump will show that the function actually specifically set a pointer value to 0 and then dereferenced it.
The location where this happens in the objdump is 0x10 bytes into a function of length 0x20 bytes.